### ISSUE TITLE
Error logs when using the processFrontmatter API

### ISSUE THREAD

**Issue by FeralFlora** [lynchjames/obsidian-writing-goals#54]
I get many instances of this error, whenever I run my property processing script that uses processFrontmatter:

```
Uncaught (in promise) SyntaxError: Unexpected end of JSON input
    at JSON.parse (<anonymous>)
    at WritingGoalsFile.loadJson (file.ts:15:21)
    at async GoalHistoryHelper.loadHistory (history.ts:39:16)
    at async GoalHistoryHelper.todaysGoalItem (history.ts:34:29)
    at async GoalHistoryHelper.updateGoalForToday (history.ts:42:20)
    at async GoalHelper.createGoal (goal-helper.ts:25:9)
    at async GoalHelper.updateGoalsFromSettings (goal-helper.ts:86:26)
    at async WritingGoals.loadNoteGoalData (plugin:writing-goals:22339:9)
    at async eval (plugin:writing-goals:22217:13)
```

See screenshot:

![image](https://github.com/lynchjames/obsidian-writing-goals/assets/33608083/28228fef-b4e5-4595-97f6-a2b83da6f571)

**Comment by FeralFlora** [comment]
While this was occuring a lot in the context of testing processFrontmatter, I'm noticing that it also happens when doing more general things like adding a heading to a note.

**Comment by FeralFlora** [comment]
Did you have a chance to look into this? These error messages are occurring constantly on my end, which makes debugging more difficult, as the console is full of these.

**Comment by Medullitus** [comment]
Is the developer looking here?? I'm getting a 'FAILED TO SHOW GOAL' error!!

**Comment by FeralFlora** [comment]
> Is the developer looking here?? I'm getting a 'FAILED TO SHOW GOAL' error!!

We saw the first time. Please don't spam the same message repeatedly.

**Comment by Medullitus** [comment]
> > Is the developer looking here?? I'm getting a 'FAILED TO SHOW GOAL' error!!
> 
> We saw the first time. Please don't spam the same message repeatedly.

Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'dailyGoalCount')
    at create_fragment$2 (plugin:writing-goals:4151:33)
    at init (plugin:writing-goals:1018:34)
    at new Simple_goal (plugin:writing-goals:4359:3)
    at FileLabels.createSimpleGoal (plugin:writing-goals:4392:9)
    at eval (plugin:writing-goals:4386:26)
    at Array.forEach (<anonymous>)
    at FileLabels.initFileLabels (plugin:writing-goals:4378:21)
    at WritingGoals.loadNoteGoalData (plugin:writing-goals:22341:29)
    at async GoalModal.onSubmit (plugin:writing-goals:20584:9)
create_fragment$2 @ plugin:writing-goals:4151
init @ plugin:writing-goals:1018
Simple_goal @ plugin:writing-goals:4359
createSimpleGoal @ plugin:writing-goals:4392
eval @ plugin:writing-goals:4386
initFileLabels @ plugin:writing-goals:4378
loadNoteGoalData @ plugin:writing-goals:22341

**Comment by Medullitus** [comment]
> > Is the developer looking here?? I'm getting a 'FAILED TO SHOW GOAL' error!!
> 
> We saw the first time. Please don't spam the same message repeatedly.

Sorry. I was just trying to find people who can help about the issue. I try to solve it but I can't do it...

**Comment by Medullitus** [comment]
> > Is the developer looking here?? I'm getting a 'FAILED TO SHOW GOAL' error!!
> 
> We saw the first time. Please don't spam the same message repeatedly.

Does your plugin work?? Can you confirm plz?? Or is it a global issue??

**Comment by Medullitus** [comment]
UPDATE: I found the issue! The HOMEPAGE plugin

When the Homepage plugin is enabled, the daily word goal plugin doesn't open and gives an error. You can disable the Homepage plugin, open the word goal window, and then reactivate the Homepage plugin. Thanks. @lynchjames  we love you and your plugin is amazing! Please don't leave us!! Thanks <3<3<3<3<3

**Comment by lynchjames** [comment]
@FeralFlora when you are running the script, are you modifying many files in the vault? For each file modification, the Writing Goals plugin checks against goals listed in plugin data JSON to work out whether to change word counts for files and/or folders. I think you are causing the plugin to hit contention with IO access to the data JSON. I should cache this info in memory to avoid high numbers of file reads, will make a change for that.

**Comment by lynchjames** [comment]
Changes included in [0.10.1](https://github.com/lynchjames/obsidian-writing-goals/releases/tag/0.10.1).
